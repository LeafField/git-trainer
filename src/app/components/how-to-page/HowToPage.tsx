import React from "react";
import Image from "next/image";
import howToImage from "../../../../public/howto.jpg";
import Button from "../button/Button";

const HowToPage = () => {
  return (
    <main className="md:px-12 lg:px-40 pb-24">
      <h1 className="text-4xl md:text-6xl xl:text-8xl font-bold text-center text-white pt-[7.5rem]">
        Git Emptyの遊び方
      </h1>
      <figure className="relative max-w-4xl mx-auto  container mt-24 md:mt-[11.25rem]">
        <Image
          src={howToImage}
          width={1605}
          height={856}
          style={{ objectFit: "cover" }}
          alt="Git Emptyの遊び方参考画像"
          className="w-full h-auto transition-all"
          sizes="100vw"
          priority
          placeholder="blur"
        />
      </figure>
      <div className="max-w-4xl mx-auto">
        <div className="pt-[7.5rem] text-base space-y-4 text-white">
          <p>
            コンソール上部にGitの練習課題が表示されるので該当するgitコマンドを入力してEnterを押してください。
          </p>
          <p>分からない時は右上の検索窓から検索してみてください。</p>
        </div>

        <div className="text-white mt-[5.62rem]">
          <h2 className="text-3xl font-bold">
            フォルダ名、リモート名などの注意
          </h2>
          <p className="text-base mt-[2.25rem]">
            リモートリポジトリのURLは「URL」と打ってください
            <br />
            例:git remote add origin URL
          </p>
          <p className="text-base mt-4">
            フォルダ名を指定する場合は「.」と打ってください
            <br />
            例:git add .
          </p>
        </div>
        <div className="w-fit mx-auto mt-[5.62rem]">
          <Button title="Let's Play!" link="/difficulty" color="primary" />
        </div>
      </div>
    </main>
  );
};

export default HowToPage;
