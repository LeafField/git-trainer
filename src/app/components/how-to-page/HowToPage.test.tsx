import { render, screen } from "@testing-library/react";
import HowToPage from "./HowToPage";

describe("HowToPage", () => {
  test("HowToPageが正しく表示されているか", () => {
    const { container } = render(<HowToPage />);
    expect(screen.getByText("Git Emptyの遊び方")).toBeInTheDocument();
    expect(
      screen.getByText(
        "コンソール上部にGitの練習課題が表示されるので該当するgitコマンドを入力してEnterを押してください。"
      )
    ).toBeInTheDocument();
    expect(
      screen.getByText("分からない時は右上の検索窓から検索してみてください。")
    ).toBeInTheDocument();
    expect(
      screen.getByText("フォルダ名、リモート名などの注意")
    ).toBeInTheDocument();

    expect(
      screen.getByRole("link", { name: "Let's Play!" })
    ).toBeInTheDocument();

    expect(container).toMatchSnapshot();
  });
});
