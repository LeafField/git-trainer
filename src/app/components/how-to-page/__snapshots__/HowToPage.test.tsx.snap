// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HowToPage HowToPageが正しく表示されているか 1`] = `
<div>
  <main
    class="md:px-12 lg:px-40 pb-24"
  >
    <h1
      class="text-4xl md:text-6xl xl:text-8xl font-bold text-center text-white pt-[7.5rem]"
    >
      Git Emptyの遊び方
    </h1>
    <figure
      class="relative max-w-4xl mx-auto  container mt-24 md:mt-[11.25rem]"
    >
      <img
        alt="Git Emptyの遊び方参考画像"
        class="w-full h-auto transition-all"
        data-nimg="1"
        decoding="async"
        fetchpriority="high"
        height="856"
        sizes="100vw"
        src="/_next/image?url=%2Fimg.jpg&w=3840&q=75"
        srcset="/_next/image?url=%2Fimg.jpg&w=640&q=75 640w, /_next/image?url=%2Fimg.jpg&w=750&q=75 750w, /_next/image?url=%2Fimg.jpg&w=828&q=75 828w, /_next/image?url=%2Fimg.jpg&w=1080&q=75 1080w, /_next/image?url=%2Fimg.jpg&w=1200&q=75 1200w, /_next/image?url=%2Fimg.jpg&w=1920&q=75 1920w, /_next/image?url=%2Fimg.jpg&w=2048&q=75 2048w, /_next/image?url=%2Fimg.jpg&w=3840&q=75 3840w"
        style="color: transparent; object-fit: cover; transition: all 0.25s ease-in-out; background-size: cover; background-position: 50% 50%; background-repeat: no-repeat;"
        width="1605"
      />
    </figure>
    <div
      class="max-w-4xl mx-auto"
    >
      <div
        class="pt-[7.5rem] text-base space-y-4 text-white"
      >
        <p>
          コンソール上部にGitの練習課題が表示されるので該当するgitコマンドを入力してEnterを押してください。
        </p>
        <p>
          分からない時は右上の検索窓から検索してみてください。
        </p>
      </div>
      <div
        class="text-white mt-[5.62rem]"
      >
        <h2
          class="text-3xl font-bold"
        >
          フォルダ名、リモート名などの注意
        </h2>
        <p
          class="text-base mt-[2.25rem]"
        >
          リモートリポジトリのURLは「URL」と打ってください
          <br />
          例:git remote add origin URL
        </p>
        <p
          class="text-base mt-4"
        >
          フォルダ名を指定する場合は「.」と打ってください
          <br />
          例:git add .
        </p>
      </div>
      <div
        class="w-fit mx-auto mt-[5.62rem]"
      >
        <a
          class="text-white block w-fit text-base md:text-2xl rounded-[2.8125rem] py-2 px-5 md:py-4 md:px-10
      font-normal hover:after:content-[""] hover:after:bg-white after:transition-all after:block
      after:h-[1px] after:w-full after:scale-0 hover:after:scale-100
      bg-primary false"
          href="/difficulty"
        >
          Let's Play!
        </a>
      </div>
    </div>
  </main>
</div>
`;
